controllers/machineset_controller.go

```diff
@@ -211,6 +211,31 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 		return ctrl.Result{}, errors.Wrap(err, "failed to list machines")
 	}
 
+	var anyDeleted bool
+	var errs []error
+	for i := range allMachines.Items {
+		machine := allMachines.Items[i]
+		if _, unhealthy := machine.GetAnnotations()[clusterv1.MachineUnhealthy]; !unhealthy {
+			continue
+		}
+
+		logger.Info("Deleting unhealthy machine", "machine", machine.GetName())
+		if err := r.Client.Delete(ctx, &machine); err != nil {
+			errs = append(errs, err)
+		}
+
+		anyDeleted = true
+	}
+	err = kerrors.NewAggregate(errs)
+
+	if err != nil {
+		logger.Info("Failed while deleting unhealthy machines", "err", err)
+		return ctrl.Result{}, errors.Wrap(err, "failed to remediate machines")
+	} else if anyDeleted {
+		// Do not do any other operations while waiting for deletion to finish
```

Comment:
can we clarify why do we need to "Do not do any other operations while waiting for deletion to finish"?